### PR TITLE
interpolation i think

### PR DESCRIPTION
**Pull request by SubAt0m1c**

### CODE REVIEW

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt [hunk 1]
```diff
@@ -205,11 +206,17 @@ object BloodCamp : Module(
...
-            val pingAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(pingPoint)
-            val endAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(endPoint)
+            val renderEndPoint = getRenderVector(endPoint, event.partialTicks, renderData.lastEndPoint)
+            val renderPingPoint = getRenderVector(pingPoint, event.partialTicks, renderData.lastPingPoint)
 
-            val time = data.time ?: return@forEach
+            renderData.lastEndPoint = endPoint
+            renderData.lastPingPoint = pingPoint
+
+            val boxOffset = Vec3((boxSize/2).unaryMinus(),1.5,(boxSize/2).unaryMinus())
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:215]: unaryminus is the same as just typing -(boxSize/2)

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt [hunk 2]
```diff
@@ -177,15 +178,15 @@ object BloodCamp : Module(
 
         val ping = ServerUtils.averagePing
 
-        forRender.filter { (entity) -> !entity.isDead }.forEach { (entity, data) ->
+        forRender.filter { (entity) -> !entity.isDead }.forEach { (entity, renderData) ->
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:181]: just use it in the filter

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt [hunk 3]
```diff
@@ -177,15 +178,15 @@ object BloodCamp : Module(
 
         val ping = ServerUtils.averagePing
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:179]: why?

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt [hunk 4]
```diff
@@ -205,11 +206,17 @@ object BloodCamp : Module(
...
+            val renderEndPoint = getRenderVector(endPoint, event.partialTicks, renderData.lastEndPoint)
+            val renderPingPoint = getRenderVector(pingPoint, event.partialTicks, renderData.lastPingPoint)
 
-            val time = data.time ?: return@forEach
+            renderData.lastEndPoint = endPoint
+            renderData.lastPingPoint = pingPoint
+
+            val boxOffset = Vec3((boxSize/2).unaryMinus(),1.5,(boxSize/2).unaryMinus())
+            val pingAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(renderPingPoint)
+            val endAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(renderEndPoint)
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:217]: you should be able to just do .offset(boxOffset + renderEndPoint), there is an operator function in vecutils for it

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt [hunk 5]
```diff
@@ -205,11 +206,17 @@ object BloodCamp : Module(
                 currVector.zCoord + speedVectors.zCoord * ping
             )
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:208]: cant you use your new function for the endPoint Vec3?

**Review by freebonsai — CHANGES_REQUESTED** [review]

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt (hunk 3 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:179]: soopy had stuff for ping so i carried that on.

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt (hunk 5 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:208]: the new function needs the endpoint vec, but i may be misunderstanding

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt (hunk 2 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:181]: not entirely sure what you mean

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt (hunk 5 above)
Comment (freebonsai) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:208]: like use the function you added to calculate it, it looks like the exact same calculation

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt (hunk 3 above)
Comment (freebonsai) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:179]: sure, but it doesnt need to be a seperate variable, just use ServerUtils.averagePing or import averagePing from it

File: odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt (hunk 2 above)
Comment (freebonsai) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:181]: forRender.filter { !it.isDead }.blahblah

Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt:181]: it doesnt work here. I think its because i use a map.

**Pull request merged**